### DISCUSSION TITLE
theory of constraints, scheduling using bottleneck control

### DISCUSSION THREAD

**Discussion by wuxuedaifu** [frePPLe/frepple#554] [Q&A] [answered]
Hi, does the community edition support the Theory of Constraints using bottleneck control, which part code is it used? thanks.

**Comment by jdetaeye** [comment, marked as answer]
Yes, the planning algorithm (in all editions of the software) supports the Theory of Constraints principle.

At a high level that means that the plan "follows the pace of the bottleneck".
Specifically;
- The algorithm doesn't allocate material if the capacity isn't available.  If a bottleneck resource is only free on day 15, then also the non-bottleneck resources, materials, preceding operations, following operations, will all be planned around day 15.
- The algorithm doesn't allocate capacity if the material isn't available.   If a bottleneck material is only available on day 20, then also the capacity, other non-bottleneck materials, preceding operations, following operations, will all be planned around day 20.
